### PR TITLE
Rework connection management, reuse HttpClient

### PR DESCRIPTION
**Pull request by evpaassen**
This pull request is primary meant to fix the `HttpConnectionManager` shutdown issue mentioned in #100. It adds a `close()` method to `ExchangeServiceBase`, which shuts down the `HttpClient`, which in turn shuts down the `HttpConnectionManager`. This required refactoring, because the `HttpClient` wasn't accessible from `ExchangeServiceBase`, only from the `HttpWebRequest` as every request created a new client. The `HttpClient` should be reused anyway for performance reasons, so it has been moved to the `ExchangeServiceBase`.

Other (related) changes:
* The `PoolingHttpClientConnectionManager` is replaced by `BasicHttpClientConnectionManager`. As `ExchangeServiceBase` is not thread-safe anyway, there's no need to use a connection pool: a basic connection manager does suffice.
* The library touched the global `CookieHandler` settings for no reason. This has been fixed.
* The JCIFS dependency has been removed, because it is unused since the migration to HttpClient 4 (#126).
* The `AutodiscoverService` did not close a response object on early return in `getLegacyUserSettingsAtUrl(...)`. The response object is now always being closed properly.
* The scope of the proxy URL was per-request, while proxy-credentials were global (allowing ony one set of proxy credentials, globally). The HttpProxyCredentials has been replaced by WebProxyCredentials, which have a per-request scope, just like the URL.
* Cookies are now handled implicitly by the HttpClient, instead of explicitly by the ourselves.

*Note: please don't squash these commits, as they fix separate (though related) issues.*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@evpaassen__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by MikeN123** [comment]
I added a commit that only allows NTLM and Basic authentication, to prevent the Kerberos warnings/slowness mentioned in #144 and #160. Basic authentication is necessary since Office 365 only seems to support Basic auth.

File: src/main/java/microsoft/exchange/webservices/data/HttpWebRequest.java [hunk 1]
```diff
@@ -220,18 +213,14 @@ public void setUrl(URL url) {
   }
 
   /**
-   * Checks if is pre authenticate.
-   *
-   * @return true, if is pre authenticate
+   * Whether to use preemptive authentication. Currently not implemented, though.
```
Comment (vboctor) [inline src/main/java/microsoft/exchange/webservices/data/HttpWebRequest.java]: Any benefit of having this while it doesn't have an effect?  I can see we have other public interface changes here, so it is not like we don't want to change the public interface.

**Comment by vboctor** [comment]
Thanks @evpaassen and @MikeN123.  Merged.

File: src/main/java/microsoft/exchange/webservices/data/HttpWebRequest.java (hunk 1 above)
Comment (evpaassen) [inline src/main/java/microsoft/exchange/webservices/data/HttpWebRequest.java]: See https://github.com/OfficeDev/ews-java-api/issues/162#issuecomment-71319928.

**Pull request merged**